src/coreclr/jit/importercalls.cpp

```diff
@@ -4717,6 +4717,13 @@ GenTree* Compiler::impSRCSUnsafeIntrinsic(NamedIntrinsic          intrinsic,
             ClassLayout*         toLayout  = nullptr;
             var_types            toType    = TypeHandleToVarType(toTypeHnd, &toLayout);
 
+            if (fromType == TYP_REF || info.compCompHnd->getBoxHelper(fromTypeHnd) == CORINFO_HELP_BOX_NULLABLE ||
+                toType == TYP_REF || info.compCompHnd->getBoxHelper(toTypeHnd) == CORINFO_HELP_BOX_NULLABLE)
```

Comment:
It is less than ideal to check for Nullable<T> by asking for box helper. This should be a new JIT/EE interface method.

Reply:
(The method should return `TypeCompareState`, e.g. similar to https://github.com/dotnet/runtime/pull/99555/files#diff-9219a33640f657e0bbc2f00f2aa40f6a5630999d8edd7e6d4f9bb79ff2ea8e03R2743 .)

Reply:
Do we even really need to block nullable here?

Reply:
Yes. We are exactly preserving the previous semantics. `where T : struct` requires the type be a non-nullable value type and thus excludes `Nullable<T>` which is considered special.

Reply:
@tannergooding does the change look correct otherwise (would it be fine assuming we "just" add the new helper for Nullable)?

Reply:
As far as I'm aware it looks correct...

However, if we're going to be adding a JIT/EE interface to handle nullable anyways, it might be nice to just have some helper to see if a given `T` meets the `where T : struct` constraint to ensure this can be more simply handled overall and would ensure we don't miss anything in the future if what generics support is expanded (if generics start supporting things like `ref T` or `T*` for example).

Reply:
Nit: The implementation rejects `default(T) is null`. It happens to match `where T : struct` today, but it is not clear whether it would continue to match `where T : struct` if generics start supporting things like `ref T` or `T*`.

Reply:
Added a `TypeCompareState isNullableType(CORINFO_CLASS_HANDLE cls)` helper now.